cmd/nerdctl/login_linux_test.go

```diff
@@ -75,3 +76,100 @@ func TestLoginWithSpecificRegHosts(t *testing.T) {
 	}
 
 }
+
+func TestLoginWithPlainHttp(t *testing.T) {
+	testutil.DockerIncompatible(t)
+	base := testutil.NewBase(t)
+	reg5000 := testregistry.NewAuthWithHttp(base, "admin", "validTestPassword", 5000, 5001)
+	reg80 := testregistry.NewAuthWithHttp(base, "admin", "validTestPassword", 80, 5002)
+	defer reg5000.Cleanup()
+	defer reg80.Cleanup()
+	testCasesForPort5000 := []struct {
+		regHost           string
+		regPort           int
+		useRegPort        bool
+		username          string
+		password          string
+		shouldSuccess     bool
+		registry          *testregistry.TestRegistry
+		shouldUseInSecure bool
+	}{
+		{
+			regHost:           "127.0.0.1",
+			regPort:           5000,
+			useRegPort:        true,
+			username:          "admin",
+			password:          "validTestPassword",
+			shouldSuccess:     true,
+			registry:          reg5000,
+			shouldUseInSecure: true,
+		},
+		{
+			regHost:           "127.0.0.1",
+			regPort:           5000,
+			useRegPort:        true,
+			username:          "admin",
+			password:          "invalidTestPassword",
+			shouldSuccess:     false,
+			registry:          reg5000,
+			shouldUseInSecure: true,
+		},
+		{
+			regHost:           "127.0.0.1",
+			regPort:           5000,
+			useRegPort:        true,
+			username:          "admin",
+			password:          "validTestPassword",
+			shouldSuccess:     false,
+			registry:          reg5000,
+			shouldUseInSecure: false,
+		},
+		{
+			regHost:           "127.0.0.1",
+			regPort:           80,
+			useRegPort:        false,
+			username:          "admin",
+			password:          "validTestPassword",
+			shouldSuccess:     true,
+			registry:          reg80,
+			shouldUseInSecure: true,
+		},
+		{
+			regHost:           "127.0.0.1",
+			regPort:           80,
+			useRegPort:        false,
+			username:          "admin",
+			password:          "invalidTestPassword",
+			shouldSuccess:     false,
+			registry:          reg80,
+			shouldUseInSecure: true,
+		},
```

Comment:
plz add a test `shouldUseInSecure: false` for `registry:  reg80` to check that there is not a regression

Reply:
OK